lib/suites.js

```diff
@@ -82,47 +82,55 @@ export function getSuiteParams({config, suiteName}) {
 }
 
 function _createEddsa2022Suite({signer}) {
-  // remove milliseconds precision
-  const date = new Date().toISOString().replace(/\.\d+Z$/, 'Z');
-  const cryptosuite = eddsa2022CryptoSuite;
   return new DataIntegrityProof({
-    signer, date, cryptosuite, legacyContext: true
+    signer,
+    date: getISODateTime(),
+    cryptosuite: eddsa2022CryptoSuite,
+    legacyContext: true
   });
 }
 
 function _createEcdsa2019Suite({signer} = {}) {
-  // remove milliseconds precision
-  const date = new Date().toISOString().replace(/\.\d+Z$/, 'Z');
-  const cryptosuite = ecdsa2019CryptoSuite;
   return new DataIntegrityProof({
-    signer, date, cryptosuite, legacyContext: true
+    signer,
+    date: getISODateTime(),
+    cryptosuite: ecdsa2019CryptoSuite,
+    legacyContext: true
   });
 }
 
 function _createEddsaRdfc2022Suite({signer}) {
-  // remove milliseconds precision
-  const date = new Date().toISOString().replace(/\.\d+Z$/, 'Z');
-  const cryptosuite = eddsaRdfc2022CryptoSuite;
-  return new DataIntegrityProof({signer, date, cryptosuite});
+  return new DataIntegrityProof({
+    signer,
+    date: getISODateTime(),
+    cryptosuite: eddsaRdfc2022CryptoSuite
+  });
 }
 
 function _createEcdsaRdfc2019Suite({signer} = {}) {
-  // remove milliseconds precision
-  const date = new Date().toISOString().replace(/\.\d+Z$/, 'Z');
-  const cryptosuite = ecdsaRdfc2019CryptoSuite;
-  return new DataIntegrityProof({signer, date, cryptosuite});
+  return new DataIntegrityProof({
+    signer,
+    date: getISODateTime(),
+    cryptosuite: ecdsaRdfc2019CryptoSuite
+  });
 }
 
-function _createEcdsaSd2023Suite({signer} = {}) {
-  // remove milliseconds precision
-  const date = new Date().toISOString().replace(/\.\d+Z$/, 'Z');
+function _createEcdsaSd2023Suite({signer, options} = {}) {
+  const mandatoryPointers = options?.mandatoryPointers ||
+    ['/issuer', '/issuanceDate'];
   const cryptosuite = createEcdsaSd2023SignCryptosuite({
-    mandatoryPointers: [
-      '/issuer',
-      '/issuanceDate'
-    ]
+    mandatoryPointers
+  });
+  const diProof = new DataIntegrityProof({
+    signer,
+    date: getISODateTime(),
+    cryptosuite
   });
-  const diProof = new DataIntegrityProof({signer, date, cryptosuite});
   diProof.proof = {id: `urn:uuid:${uuid()}`};
   return diProof;
 }
+
+function getISODateTime(date = new Date()) {
```

Comment:
To follow DB internal-only function helper style:

```suggestion
function _getISODateTime(date = new Date()) {
```